guacamole-docker/README.md

```diff
@@ -55,31 +55,31 @@ Deploying Guacamole with PostgreSQL authentication
 --------------------------------------------------
 
     docker run --name some-guacamole --link some-guacd:guacd \
-        --link some-postgres:postgres      \
-        -e POSTGRES_DATABASE=guacamole_db  \
-        -e POSTGRES_USER=guacamole_user    \
-        -e POSTGRES_PASSWORD=some_password \
-        -e POSTGRES_DATABASE_FILE=/run/secrets/<secret_name> \
-        -e POSTGRES_USER_FILE=/run/secrets/<secret_name> \
-        -e POSTGRES_PASSWORD_FILE=/run/secrets/<secret_name> \
+        --link some-postgresql:postgresql      \
```

Comment:
This particular line may actually need to not be changed - my guess is that `postgres` is the default name of the container for PostgreSQL databases. Also, this doesn't really match up to any `guacamole.properties` entries.

@mike-jumper Thoughts on this one?

Reply:
Yes, I've hesitated too when doing the changes but I thought it may be less error-prone if everything is named postresql instead of postgres only for the link. I can revert this part if you prefer.
What about the username variable ?

Reply:
> This particular line may actually need to not be changed ... Thoughts on this one?

Hm ... I'm not sure off-hand. Most of my Docker deployments have been docker-compose for a while, and I recall someone mentioning that Docker had deprecated `--link`.

I think the "postgresql" part is used by Docker to generate environment variables, and so would need to match whatever the container uses.

Reply:
Originally, the `POSTGRES_*` prefix was chosen despite the property names to match the similar environment variables used by Docker's official `postgres` container. Aligning things universally with properties makes sense and would allow some automation for setting properties (via the environment properties feature).

There will be pain for anyone using the old variables. If we do this, it's probably worth adding compatibility for the `POSTGRES_` prefix and `--postgres` option.

Reply:
`--link` is deprecated according to this [Legacy Container links](https://docs.docker.com/network/links/) but it's still working.  No sure how many people are using it.
If it should be removed, it should be removed in another issue/PR.

The `--link some-postgresql:postgresql ` means that environment variables for the ports are generated using the `<name>_PORT_<port>_<protocol>` format where `<name>` is the name of the last part of the link (`postgresql`) for us but `postgres` before my change.

We can add compatibility for older environment variable names but at some point it will be removed and will break for some people, maybe it's "easier" to add it in compatibility change when the image will be released so people can see it explicitely. They'll need to do this change at some point if we want to align everything.

If we don't want to align everything, we must solve the otherway and rename the few environment variables using `POSTGRESQL_*` to `POSTGRES_*` because it doesn't make sense to have some using one syntax and other using the other.

Reply:
Yes `--link` is deprecated but can still be used. If you want to remove them I guess it will be done in another issue/PR. See more on [Legacy container links](https://docs.docker.com/network/links/). The environment variables related to the link are generated from the last part of the `--link mypostgres:postgres` which results in `<name>_PORT_<port>_<protocol>` variable and `<name>_NAME` variable. So in our case `postgres`. In the PR I've renamed it too so it's consistent with other variable names but it's just to make it prettier.

For the other variables, renaming them from `POSTGRES_` to `POSTGRESSQL_` will of course break current deployment. If needed we can have both but I can imagine it will be removed one day and it will cause the same issue. It would maybe be easier to point it in the release notes that the variables changed their names and need to be changed in docker compose etc.
How do you want to move forward on it ?

Reply:
@jbpaux I agree that removing link is different in scope from this, so no need to worry about that.

For the renaming, I would think something simple at the beginning of the script that checks for the old variables and assigns the values to the new variables would probably work? That way the remaining checks, here, would only have to consider a single set of variables - the "correctly" named ones.

Reply:
I did it, it's long but should work as expected 😅